xbmc/addons/addoninfo/AddonType.cpp

```diff
@@ -23,6 +23,30 @@ static const std::set<AddonType> dependencyTypes = {
 
 using namespace ADDON;
 
+CAddonType::CAddonType(const std::string& type) : m_type(AddonType::UNKNOWN)
+{
+  if (!type.empty())
+  {
+    try
+    {
+      auto val = std::stoi(type);
```

Comment:
do not use `auto` for fondumental type vars
the use of `stoi` may raise unhandled exceptions
use `std::from_chars` instead